### DISCUSSION TITLE
How do I combine x-data variables into one?

### DISCUSSION THREAD

**Discussion by JF-626164** [alpinejs/alpine#3689] [1. Help]
I'm trying to figure out how to combine multiple x-data variables into one. For example, I have 3 simple numeric input fields, I have input1-3 in the x-data and the using x-model to get the value of whats inputted in the respective fields. 

What I want to do is, on a button click, I want to combine those three into one and pass that one value to a function.

**Comment by JF-626164** [comment]
I believe I've figured out a workaround, in a x-on:click, I can add the variables together in it.
Its interesting how working for hours on this and then shortly after I ask for help I figure it out.

**Comment by ekwoka** [comment]
```html
<div x-data="{ numbers: [] }">
  <input type="number" x-model.number="numbers[0]" />
  <button @click="func(numbers.reduce((a,b) => a + b)"></button>
</div>
```
